src/command_modules/azure-cli-role/azure/cli/command_modules/role/commands.py

```diff
@@ -77,42 +85,43 @@ def load_command_table(self, _):
         g.custom_command('create', 'create_role_assignment')
         g.custom_command('list-changelogs', 'list_role_assignment_change_logs')
 
-    with self.command_group('ad app', client_factory=get_graph_client_applications, resource_type=PROFILE_TYPE, min_api='2017-03-10') as g:
+    with self.command_group('ad app', client_factory=get_graph_client_applications, resource_type=PROFILE_TYPE,
+                            min_api='2017-03-10', exception_handler=graph_err_handler) as g:
         g.custom_command('create', 'create_application')
         g.custom_command('delete', 'delete_application')
         g.custom_command('list', 'list_apps')
-        g.custom_command('show', 'show_application', exception_handler=empty_on_404)
```

Comment:
`empty_on_404` never works as it only captures CloudError, but all graph commands raises `GraphErrorException`